Do I need to add -lpthread to linker flags?

## Platform

  - SeqAn version: 3.0.0
  - Operating system: Linux weissweiler.molgen.mpg.de 4.19.19.mx64.244 #1 SMP Tue Feb 5 13:01:13 CET 2019 x86_64 GNU/Linux
  - Compiler: gcc (GCC) 7.3.0

## Description

### How to repeat the problem

I'm writing an application based on seqan3. This is how my project directory looks like:
```
submodules/seqan3
src/
- CMakeLists.txt
- main.cpp
```

CMakeLists.txt:
```
cmake_minimum_required (VERSION 3.4)
project (seqan3_tutorial CXX)
set(SeqAn3_DIR "${CMAKE_SOURCE_DIR}/../submodules/seqan3/build_system")
find_package (SeqAn3 REQUIRED)
add_executable (main main.cpp)
target_link_libraries (main seqan3::seqan3)
```

main.cpp:
```
#include <seqan3/argument_parser/all.hpp>   // includes all necessary headers
#include <seqan3/core/debug_stream.hpp>     // our custom output stream
#include <seqan3/std/filesystem>            // use std::filesystem::path
#include <seqan3/io/sequence_file/all.hpp>  // FASTA support
#include <seqan3/io/alignment_file/all.hpp> // SAM/BAM support

using namespace seqan3;

struct cmd_arguments
{
    std::filesystem::path alignment_file_path{};
    std::filesystem::path reference_file_path{};
};

void initialize_argument_parser(argument_parser & parser, cmd_arguments & args)
{
    parser.info.author = "foo";
    parser.info.short_description = "foo";
    parser.info.version = "0.0.1";
    parser.add_positional_option(args.alignment_file_path, "Read alignments in SAM or BAM format.",
                                 input_file_validator{{"sam", "bam"}} );
    parser.add_positional_option(args.reference_file_path, "Reference genome in FASTA format",
                                 input_file_validator{{"fa", "fasta"}} );
}

int main(int argc, char ** argv)
{
    argument_parser myparser{"detectBreakends", argc, argv};        // initialise myparser
    cmd_arguments args{};
    initialize_argument_parser(myparser, args);
    try
    {
        myparser.parse();                                          // trigger command line parsing
    }
    catch (parser_invalid_argument const & ext)                     // catch user errors
    {
        debug_stream << "[Error] " << ext.what() << "\n"; // customise your error message
        return -1;
    }
    // run_program(args.alignment_file_path, args.reference_file_path);
    return 0;
}

```

### Expected behaviour

I would expect the program to compile successfully with:
```
mkdir build
cd build
cmake ../src
make
```

### Actual behaviour

```
>make
Scanning dependencies of target main
[ 50%] Building CXX object CMakeFiles/main.dir/main.cpp.o
[100%] Linking CXX executable main
/usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/main.dir/main.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/main.dir/build.make:86: main] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/main.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

I found that add -lpthread to the linker flags solves this issue. But I wondered why it is necessary to do this manually if I only use the seqan argument parser and not much more. Any idea?

Cheers
David

--- a/build_system/seqan3-config.cmake
+++ b/build_system/seqan3-config.cmake
@@ -363,6 +363,24 @@ else ()
 endif ()
 
 # ----------------------------------------------------------------------------
+# thread support (pthread, windows threads)
+# ----------------------------------------------------------------------------
+
+set (THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package (Threads QUIET)
+
+if (Threads_FOUND)
+    set (SEQAN3_LIBRARIES ${SEQAN3_LIBRARIES} Threads::Threads)
+    if ("${CMAKE_THREAD_LIBS_INIT}" STREQUAL "")
+        seqan3_config_print ("Thread support:             builtin.")
+    else ()
+        seqan3_config_print ("Thread support:             via ${CMAKE_THREAD_LIBS_INIT}")
+    endif ()
+else ()
+    seqan3_config_print ("Thread support:             not found.")
+endif ()
+
+# ----------------------------------------------------------------------------
 # Require Ranges and SDSL
 # ----------------------------------------------------------------------------
 

